tern/utils/general.py

```diff
@@ -112,25 +112,40 @@ def split_command(shell_script):
 def parse_shell_variables_and_command(concatenated_command):
     '''given a concatenated command, classify the variable and command type,
     and then parse it '''
+    # append white space at the end since we are matching white space for
+    # the end of an assignment, and this white space will be removed by
+    # clean_command()
+    concatenated_command += ' '
     # pattern for matching variable, looking for '='
-    assignment_pattern = r'^([A-Za-z_][A-Za-z0-9_]*)=(.*)'
+    # the second group first macthes a string assignment, if not, it will macth
+    # until it first meets a white space
+    assignment_pattern = r'([A-Za-z_][A-Za-z0-9_]*)=((".*")|(.*?)(?=[\s]))'
     export_pattern = r'^export ([A-Za-z_][A-Za-z0-9_]*)=(.*)'
-    variable_pattern = assignment_pattern + r'|' + export_pattern
-    match_res = re.match(variable_pattern, concatenated_command)
+    match_export = re.match(export_pattern, concatenated_command)
+    macth_assignment = re.finditer(assignment_pattern, concatenated_command)
```

Comment:
Suggest fixing the spelling: `match_assignment`.

Reply:
Fixed!